### PR TITLE
beginning tests, pushing up for visibility

### PR DESCRIPTION
**Pull request by stephnr**
Feel free to contribute everyone. I started us off with jasmine tests.

To run the tests, just open index.html and Jasmine will run them.

### CODE REVIEW

**Comment by kitten** [comment]
We could also try to utilise [Jest](https://github.com/facebook/jest). I used it in some projects and it's really nice

**Comment by stephnr** [comment]
@philplckthun: I will look into this. Currently Jasmine is being used. I will provide my assessment soon.

**Comment by stephnr** [comment]
@philplckthun I like it. I would suggest we only use it because of how it integrates a virtual dom for CLI support on running the tests. Do you mind configuring the current branch here to work with Jest? 

Please make changes on my forked repo instead of the bendc/sprint repo

**Comment by kitten** [comment]
@Stephn-R I can look into it, but I'd prefer to start from scratch actually ^^

**Comment by stephnr** [comment]
No worries @philplckthun. Already done.

**Comment by kitten** [comment]
LGTM :ship: 

And just wanted to note, that this depends on PR #21 as well.

**Comment by stephnr** [comment]
@philplckthun Can you explain why that is?

File: sprint.js [hunk 1]
```diff
@@ -1405,3 +1405,5 @@ var Sprint;
     window.$ = Sprint
   }
 }());
+
+module.exports = $;
```
Comment (kitten) [inline sprint.js:1409]: insufficient

**Comment by kitten** [comment]
I added a note to line 1409. There you added a simple module export statement to make sprint usable as a module for Jest. This is insufficient though, as this breaks normal script loading of the js file. It would throw: "Uncaught ReferenceError: module is not defined".

My PR #21 enables a module pattern for AMD modules, CommonJS (What you tried to add) and browser globals. So #21 is a dependency of this PR.

**Comment by stephnr** [comment]
:+1:

**Comment by rossPatton** [comment]
Is this the most recent pr for unit testing? Need help writing tests?

**Comment by stephnr** [comment]
@rossPatton yes, feel free to chime in and write some tests. Make sure your commits are made on the branch associated with this PR

**Comment by kitten** [comment]
@Stephn-R Can you open a new PR for the fork? It's maybe a good idea to rebase it, so the line 1409 issue is resolved immediately.

https://github.com/philplckthun/sprint

**Pull request closed without merging**